providers/song-info-front.js

```diff
@@ -55,25 +55,62 @@ module.exports = () => {
 			data.videoDetails.isPaused = false;
 			ipcRenderer.send("video-src-changed", JSON.stringify(data));
 		}
-	}, { once: true, passive: true });
+	}, {once: true, passive: true});
 };
 
 function setupTimeChangeListener() {
 	const progressObserver = new MutationObserver(mutations => {
 		ipcRenderer.send('timeChanged', mutations[0].target.value);
 		global.songInfo.elapsedSeconds = mutations[0].target.value;
 	});
-	progressObserver.observe($('#progress-bar'), { attributeFilter: ["value"] })
+	progressObserver.observe($('#progress-bar'), {attributeFilter: ["value"]})
 }
 
 function setupRepeatChangeListener() {
 	const repeatObserver = new MutationObserver(mutations => {
-		ipcRenderer.send('repeatChanged', mutations[0].target.title);
+		document.cookie.split(';').forEach((cookie) => {
+			let cookieName = cookie.substring(0, cookie.indexOf("=")).replaceAll(" ", "");
+			if (cookieName === 'PREF') {
+				let value = cookie.replace(cookieName + "=", "").replaceAll(" ", "");
+				value.split('&').forEach((pair) => {
+					if (pair !== '') {
+						let splitpair = pair.split('=');
+						if (splitpair[0] === "repeat") {
+							if (splitpair[1] === "NONE")
+								ipcRenderer.send('repeatChanged', "Repeat off");
+							else if (splitpair[1] === "ONE") //MPRIS Playlist and Track Codes are switched to look the same as yt-music icons
+								ipcRenderer.send('repeatChanged', "Repeat one");
+							else if (splitpair[1] === "ALL")
+								ipcRenderer.send('repeatChanged', "Repeat all");
+						}
+					}
+				});
+			}
+		});
```

Comment:
Instead of all this cookie stuff, try:
```js
ipcRenderer.send('repeatChanged', $('ytmusic-player-bar').getState().queue.repeatMode)
```

Reply:
Your short version works, I'd still choose to send "Repeat off" | "Repeat one" | "Repeat all" since that was what was sent before I touched the function and I dont know if any other plugins also listen for 'repeatChanged'. I'd hate to break those.

Reply:
>  I dont know if any other plugins also listen for 'repeatChanged'

For now there aren't any, that's why I proposed moving code specific to MPRIS to its own file

>  since that was what was sent before I touched the function

Yeah well doesn't mean that it's right 😅